Ical.Net/Evaluation/Evaluator.cs

```diff
@@ -43,7 +43,7 @@ protected void IncrementDate(ref CalDateTime dt, RecurrencePattern pattern, int
                 case FrequencyType.Yearly:
                     // When a rule uses BYWEEKNO, recurrence enumeration
                     // is based on week numbers relative to the year.
-                    // So we preserve January 1st when adding years; otherwise, preserve month/day.
+                    // So we preserve the weekday when using BYWEEKNO and preserve month/day otherwise.
```

Comment:
oh, I'm sorry, I guess I was wrong with my comment in #513. We indeed preserve Jan 1st, not the weekday. So I guess the previous version was better.